test/test_datapoints.py

```diff
@@ -203,4 +203,3 @@ def test_deepcopy(datapoint, requires_grad):
 
     assert type(datapoint_deepcopied) is type(datapoint)
     assert datapoint_deepcopied.requires_grad is requires_grad
-    assert datapoint_deepcopied.is_leaf
```

Comment:
the deepcopy isn't a leaf anymore because it went through `wrap_like()`, so it's got an "ancestor".
I don't think `is_leaf` is part of the `deepcopy` contract anyway? I don't think we really need to enforce this.

Reply:
> the deepcopy isn't a leaf anymore because it went through `wrap_like()`, so it's got an "ancestor". I don't think `is_leaf` is part of the `deepcopy` contract anyway? I don't think we really need to enforce this.

I don't think it is specified anywhere, so I'm ok with removing this check. Might be surprising to users though if they bank on this. Let's find out though :shrug: